demo/site/src/news/NewsPage.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+import { DamImageBlock } from "@src/common/blocks/DamImageBlock";
+import { fetcher } from "@src/util/Fetcher";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+import { useState } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+import { GQLNewsIndexPageQuery } from "./NewsPage.loader.generated";
+
+type News = GQLNewsIndexPageQuery["newsList"];
+
+export function NewsPage({ initialData, limit }: { initialData: News; limit: number }) {
+    const [pages, setPages] = useState<number[]>([]); // Contains array of offsets
```

Comment:
a SEO friendly approach would somehow have a page or similar in the URL (maybe out of scope for this PR)